docs/docs/contribute.md

```diff
@@ -1 +1,176 @@
+at least Python 3.8.
+
+**Step 6**: Add the upstream remote. This saves a reference to the main `sbi` repository,
+which you can use to keep your repository synchronized with the latest changes:
+```
+git remote add upstream git@github.com:sbi-dev/sbi.git
+```
+Check that the upstream and origin remote aliases are configured correctly by
+running `git remote -v` which should display:
+```
+origin  git@github.com:$USERNAME/sbi.git (fetch)
+origin  git@github.com:$USERNAME/sbi.git (push)
+upstream        git@github.com:sbi-dev/sbi.git (fetch)
+upstream        git@github.com:sbi-dev/sbi.git (push)
+```
+
+**Step 7**: Install `pre-commit` to run code style checks before each commit:
+```
+pip install pre-commit
+pre-commit install
+```
+
+You should now have a working installation of `sbi` and a git repository
+properly configured for making contributions. The following steps describe the
+process of modifying code and submitting a pull request:
+
+**Step 8**: Synchronize your main branch with the upstream/main branch. See more details on [GitHub Docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork):
+```
+git checkout main
+git fetch upstream
+git merge upstream/main
+```
+
+**Step 9**: Create a feature branch to hold your development changes:
+```
+git checkout -b my_feature
+```
+and start making changes. Always use a feature branch! It’s good practice
+to never work on the main branch, as this allows you to easily get back to a
+working state of the code if needed (e.g., if you’re working on multiple
+changes at once, or need to pull in recent changes from someone else to get
+your new feature to work properly). In most cases you should make PRs into the
+upstream’s main branch.
+
+**Step 10**: Develop the feature on your feature branch on your computer, using
+Git to do the version control. When you’re done editing, add changed files
+using `git add` and then `git commit`:
+```
+git add modified_files
+git commit -m "description of your commit"
+```
+to record your changes in Git, then push the changes to your GitHub account with:
+```
+git push -u origin my_feature
+```
+The `-u` flag ensure that your local branch will be automatically linked with the remote branch, so you can later use `git push` and `git pull` without any extra arguments.
+
+**Step 11**: Follow [these](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) instructions to create a pull request from your fork.
+This will send a notification to `sbi` maintainers and trigger reviews and comments
+regarding your contribution.
+
+!!! note
+    It is often helpful to keep your local feature branch synchronized
+    with the latest changes of the main `sbi` repository:
+    ```
+    git fetch upstream
+    git merge upstream/main
+    ```
+
+### Style conventions and testing
```

Comment:
Please make sure that you also include changes which happened on main since you started working on it.

Reply:
OK, I have rebased the branch to current main status